clab/clab.go

```diff
@@ -10,19 +10,20 @@ import (
 	"time"
 
 	"github.com/containernetworking/plugins/pkg/ns"
-	"github.com/docker/docker/api/types"
-	docker "github.com/docker/docker/client"
 	log "github.com/sirupsen/logrus"
+	"github.com/srl-labs/containerlab/runtime"
+	"github.com/srl-labs/containerlab/types"
+	"github.com/srl-labs/containerlab/utils"
 )
 
 type CLab struct {
-	Config       *Config
-	TopoFile     *TopoFile
-	m            *sync.RWMutex
-	Nodes        map[string]*Node
-	Links        map[int]*Link
-	DockerClient *docker.Client
-	Dir          *Directory
+	Config   *Config
+	TopoFile *TopoFile
+	m        *sync.RWMutex
+	Nodes    map[string]*types.Node
+	Links    map[int]*Link
```

Comment:
if we move types to their own package, should the `Link` type be moved as well?

Reply:
yeah, perhaps. I was trying to minimize the number of changes. if you think it makes sense as a part of this PR, i'll move it.

Reply:
fixed in f000c0f